samples/cpp/train_HOG.cpp

```diff
@@ -345,7 +349,7 @@ int main( int argc, char** argv )
         clog << "...[done]" << endl;
 
         labels.clear();
-        labels.assign( pos_lst.size(), +1 );
+        labels.assign( positive_count, +1 );
         labels.insert( labels.end(), neg_lst.size(), -1);
```

Comment:
If there are small images in the negative set they will be discarded during `computeHOG` so the training set will be shorter than `neg_lst.size()`. Could you please use evaluation approach similar to the one in the first training step

Reply:
Other parts of the code looks fine for me. It would be great to describe details of improved sample behavior